src/app/pipelines/components/pipelines-editor/pipelines-editor.component.ts

```diff
@@ -822,14 +822,23 @@ export class PipelinesEditorComponent implements OnInit {
           step.params.forEach((param) => {
             if (
               param.required &&
-              (!param.value ||
-                (param.type !== 'object' && param.value.trim().length === 0))
+              (param.defaultValue.length === 0 ||
+                typeof param.defaultValue === undefined ||
+                (param.type !== 'object' &&
+                  param.value &&
+                  param.value.trim().length === 0))
             ) {
-              errors.push(
-                `Step ${step.id} has a required parameter ${param.name} that is missing a value.`
-              );
+              if (!param.value) {
+                errors.push(
+                  `Step ${step.id} has a required parameter ${param.name} that is missing a value.`
+                );
+              }
             }
-            if (param.value && param.value.endsWith('&')) {
+            if (
+              param.value &&
+              typeof param.value === 'string' &&
+              param.value.endsWith('&')
```

Comment:
Should be startsWith instead of endsWith.

Reply:
I am using endsWith to check if the value for a pipeline is empty. As you can see in the image if I do not have selected a pipeline, the value for that parameter is '&'. I am doing this to validate and ensure that the user has selected a pipeline and display the error message.
![image](https://user-images.githubusercontent.com/28541154/76253300-0d475980-6210-11ea-8554-b266f0da1f51.png)